packages/i18n-utils/src/localize-url.tsx

```diff
@@ -169,10 +169,22 @@ export const urlLocalizationMapping: UrlLocalizationMapping = {
 	},
 };
 
+function hasTrailingSlash( urlString: string ) {
+	let url;
+	try {
+		url = new URL( String( urlString ), INVALID_URL );
+	} catch ( e ) {
+		return false;
+	}
+
+	return url.pathname.endsWith( '/' );
```

Comment:
Nit: I might be missing something, but we should be able to just use a constant and return immediately, no?

```suggestion
	try {
		const url = new URL( String( urlString ), INVALID_URL );
		return url.pathname.endsWith( '/' );
	} catch ( e ) {
		return false;
	}
```

Reply:
Nope, apparently I was the one missing something 😄 Done!